### PR TITLE
feat(snowflake): parse CREATE SEQUENCE

### LINKED ISSUE #2954: support parsing sequence related statements
**Is your feature request related to a problem? Please describe.**
When using fakesnow for unit testing we discovered that `CREATE SEQUENCE seq1` is currently not supported by sqlglot. As is accessing sequences.

**Describe the solution you'd like**
It would be terrific it sqlglot could also transpile sequence creation, access and deletion. 

**Describe alternatives you've considered**
We have currently mocked the parts of our code that relied on the sequences being present. Since the sequence is only created implicitly via SQLalchemy, we also considered changing the table definitions during testing.

**Additional context**
at least between duckdb and snowflake there are bigger differences in how to access a sequence vs creating one:
duckdb: `select nextval(seq1)`
snowflake: `select seq1.nextval`

### PR DESCRIPTION
**Pull request by tekumara**
see https://docs.snowflake.com/en/sql-reference/sql/create-sequence

partially resolves #2954

### CODE REVIEW

File: sqlglot/expressions.py [hunk 1]
```diff
@@ -3619,6 +3621,14 @@ def output_name(self) -> str:
         return self.name
 
 
+class Increment(Expression):
+    arg_types = {"this": True}
+
+
+class Start(Expression):
+    arg_types = {"this": True}
+
```
Comment (georgesittas) [inline sqlglot/expressions.py:3630]: We don't need these, right? You can just store the expressions as is in the corresponding arguments.

Btw, you could also do `pass` here.

File: sqlglot/generator.py [hunk 2]
```diff
@@ -1541,6 +1547,14 @@ def historicaldata_sql(self, expression: exp.HistoricalData) -> str:
         expr = self.sql(expression, "expression")
         return f"{this} ({kind} => {expr})"
 
+    def start_sql(self, expression: exp.Start) -> str:
+        this = self.sql(expression, "this")
+        return f"START WITH {this}"
+
+    def increment_sql(self, expression: exp.Increment) -> str:
+        this = self.sql(expression, "this")
+        return f"INCREMENT BY {this}"
```
Comment (georgesittas) [inline sqlglot/generator.py:1556]: So we'd get rid of these as well and do this in `create_sql`.

**Comment by georgesittas** [comment]
Hey @tekumara, this may be good for now but we need to ensure `CREATE SEQUENCE` doesn't fail to parse in other dialects. E.g. right now we accept the following Oracle statement, but parse it as a `Command`:

```sql
CREATE SEQUENCE customers_seq
 START WITH     1000
 INCREMENT BY   1
 NOCACHE
 NOCYCLE;
```

Please check & test a few more syntax specs that aren't covered by this PR, as well as the remaining clauses in Snowflake's syntax. For example, I believe the parser still doesn't recognize the `=`, the preceding `WITH` and the `ORDER` / `NOORDER` keywords. We should also make sure these are still handled gracefully.

Some references I found that may be useful:
- https://docs.oracle.com/en/database/oracle/oracle-database/19/sqlrf/CREATE-SEQUENCE.html#GUID-E9C78A8C-615A-4757-B2A8-5E6EFB130571
- https://www.postgresql.org/docs/current/sql-createsequence.html
- https://learn.microsoft.com/en-us/sql/t-sql/statements/create-sequence-transact-sql?view=sql-server-ver16
- https://duckdb.org/docs/sql/statements/create_sequence.html

**Comment by VaggelisD** [comment]
Hi @tekumara,

Thanks for your contribution! It's now merged and in a follow up iteration of mine I'll address @georgesittas 's comments in order to test & generalize `CREATE SEQUENCE` across dialects if you want to follow this issue.

**Pull request merged**